lib/rucio/daemons/badreplicas/necromancer.py

```diff
@@ -15,180 +15,170 @@
+from rucio.db.sqla.constants import ReplicaState
 from rucio.common import exception
 from rucio.common.cache import make_region_memcached
 from rucio.common.config import config_get, config_get_int
 from rucio.common.exception import DatabaseException
-from rucio.common.logging import formatted_logger, setup_logging
-from rucio.common.utils import daemon_sleep
-from rucio.core import monitor, heartbeat
+from rucio.common.logging import setup_logging
+from rucio.core import monitor
 from rucio.core.replica import list_bad_replicas, get_replicas_state, get_bad_replicas_backlog
 from rucio.core.rule import (update_rules_for_lost_replica, update_rules_for_bad_replica,
                              get_evaluation_backlog)
-from rucio.db.sqla.constants import ReplicaState
+from rucio.daemons.common import run_daemon
 
-GRACEFUL_STOP = threading.Event()
+if TYPE_CHECKING:
+    from rucio.daemons.common import HeartbeatHandler
+
+graceful_stop = threading.Event()
 REGION = make_region_memcached(expiration_time=config_get_int('necromancer', 'cache_time', False, 600))
 
 
-def necromancer(thread=0, bulk=5, once=False, sleep_time=60):
+def necromancer(bulk: int, once: bool = False, sleep_time: int = 60) -> None:
     """
     Creates a Necromancer Worker that gets a list of bad replicas for a given hash,
     identify lost DIDs and for non-lost ones, set the locks and rules for reevaluation.
 
-    :param thread: Thread number at startup.
     :param bulk: The number of requests to process.
     :param once: Run only once.
     :param sleep_time: Thread sleep time after each chunk of work.
     """
+    run_daemon(
+        once=once,
+        graceful_stop=graceful_stop,
+        executable='necromancer',
+        logger_prefix='necromancer',
+        partition_wait_time=1,
+        sleep_time=sleep_time,
+        run_once_fnc=functools.partial(
+            run_once,
+            bulk=bulk,
+        ),
+    )
+
+
+def run_once(heartbeat_handler: "HeartbeatHandler", bulk: int, **_kwargs) -> bool:
+    worker_number, total_workers, logger = heartbeat_handler.live()
+    logger(logging.INFO, 'Graceful stop done')
+    must_sleep = True
+
+    # Check if there is a Judge Evaluator backlog
+    try:
+        max_evaluator_backlog_count = config_get('necromancer', 'max_evaluator_backlog_count')
+        max_evaluator_backlog_count = int(max_evaluator_backlog_count)
+    except (NoOptionError, NoSectionError, RuntimeError, ValueError):
+        max_evaluator_backlog_count = None
+    try:
+        max_evaluator_backlog_duration = config_get('necromancer', 'max_evaluator_backlog_duration')
+        max_evaluator_backlog_duration = int(max_evaluator_backlog_duration)
+    except (NoOptionError, NoSectionError, RuntimeError, ValueError):
+        max_evaluator_backlog_duration = None
+    if max_evaluator_backlog_count or max_evaluator_backlog_duration:
+        evaluator_backlog_count, evaluator_backlog_duration = get_evaluation_backlog(expiration_time=60)
```

Comment:
why does it change here ? maybe pass it via adding a new parameter to run_once ? like expiration_time = sleep_time